src/value.cpp

```diff
@@ -977,11 +977,9 @@ namespace Exiv2 {
         // sprintf wants to add the null terminator, so use oversized buffer
         char temp[9];
 
-        int wrote = sprintf(temp, "%04d%02d%02d",
-                            date_.year, date_.month, date_.day);
-        assert(wrote == 8);
```

Comment:
To be honest, I'd rather keep the assertion. Just in case someone changes the `sprintf` and something else relies on this function returning 8. If the check is truly pointless, then the compiler will optimize it away (I doubt it'll do that in this case though).

Reply:
Yes, I thought about keeping it since I am anyways using now the variable `wrote` in the next two lines. I will add it back.

The check will be pointless in release mode, where the assert calls will not generate any code at all. That's why we were getting the warnings in first place.